pat/pat.go

```diff
@@ -283,6 +284,31 @@ func (p *Pattern) String() string {
 }
 
 /*
+URLPath replaces the named matches in the Pattern with the values provided
+and returns the resulting string. It can be used to generate a URL
+that would match the given pattern with the provided variables. You must
+provide a mapping for all named matches defined in the Pattern.
+*/
+func (p *Pattern) URLPath(vars map[pattern.Variable]string) (string, error) {
```

Comment:
One other thing: this takes `map[pattern.Variable]string`, while much of the rest of the `pattern` ecosystem produces and consumes `map[pattern.Variable]interface{}`. I know it's kind of ugly (and if I end up regretting one thing from this redesign I feel like it'll probably be that type), but we should probably accept that type here as well (and type assertion our way to victory)